internal/printer/print-to-js.go

```diff
@@ -686,82 +690,79 @@ func handleSlots(p *printer, n *Node, opts RenderOptions, depth int) {
 			}
 		}
 		if c.Expression {
-			nestedSlotsCount := 0
-			var firstNestedSlotProp string
+			// Only slot ElementNodes (except expressions containing only comments) or non-empty TextNodes!
+			// CommentNode, JSX comments and others should not be slotted
+			if expressionOnlyHasComment(c) {
+				continue
+			}
+			nestedSlotsInExprCount := 0
+			hasAnyDynamicSlotsInExpr := false
```

Comment:
This is what the fix is all about: localizing these states per expression. Previously we used a global variables to track them, which caused leaks in other expressions.

The rest is just polishing and refactoring

Reply:
Should we add a test with comments too?

Reply:
Ah, I should've clarified that we already have tests for that. I just moved the code above to do the check earlier